File: VB/GridView.FilteringRow/Controllers/GridViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports GridView.FilteringRow.Models

Namespace GridView.FilteringRow.Controllers
	<HandleError> _
	Public Class GridViewController
		Inherits Controller
		Public Function Index() As ActionResult
			Return FilteringRow()
		End Function
		Public Function FilteringRow() As ActionResult
			ViewData("ShowFilterRow") = True
			ViewData("ShowFilterRowMenu") = True
			Return View("FilteringRow", NorthwindDataProvider.GetInvoices())
		End Function
		Public Function FilteringRowPartial() As ActionResult
			GetRequestParemetrs()
			Return PartialView("FilteringRowPartial", NorthwindDataProvider.GetInvoices())
		End Function

		Private Sub GetRequestParemetrs()
			ViewData("ShowFilterRow") = If(String.IsNullOrEmpty(Request.Params("ShowFilterRow")), True, Boolean.Parse(Request.Params("ShowFilterRow")))
			ViewData("ShowFilterRowMenu") = If(String.IsNullOrEmpty(Request.Params("ShowFilterRowMenu")), True, Boolean.Parse(Request.Params("ShowFilterRowMenu")))
		End Sub
	End Class
End Namespace

File: CS/GridView.FilteringRow/Controllers/GridViewController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using GridView.FilteringRow.Models;

namespace GridView.FilteringRow.Controllers {
    [HandleError]
    public class GridViewController : Controller {
        public ActionResult Index() {
            return FilteringRow();
        }
        public ActionResult FilteringRow() {
            ViewData["ShowFilterRow"] = true;
            ViewData["ShowFilterRowMenu"] = true;
            return View("FilteringRow", NorthwindDataProvider.GetInvoices());
        }
        public ActionResult FilteringRowPartial() {
            GetRequestParemetrs();
            return PartialView("FilteringRowPartial", NorthwindDataProvider.GetInvoices());
        }

        void GetRequestParemetrs() {
            ViewData["ShowFilterRow"] = string.IsNullOrEmpty(Request.Params["ShowFilterRow"]) ? true : bool.Parse(Request.Params["ShowFilterRow"]);
            ViewData["ShowFilterRowMenu"] = string.IsNullOrEmpty(Request.Params["ShowFilterRowMenu"]) ? true : bool.Parse(Request.Params["ShowFilterRowMenu"]);
        }
    }
}

